sec_certs/bsi_parse.py

```diff
@@ -0,0 +1,140 @@
+import re
+import time
+
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
+from serialization import ComplexSerializableType, CustomJSONDecoder, CustomJSONEncoder
+
+"""
+architecture of the BSI page we should use:
+    -The table with class "textualData link smallerFont"
+    -1st <td> of each rows contains the link to the product's page and has a class
+    -the link is in a <a> element
+    
+    What I want to do:
+    Get a list of all <a> elements that have a <td class = * white-space-nowrap> as parent
+"""
+
+"""
+https://www.bsi.bund.de/EN/Topics/Certification/certified_products/Archiv_reports.html?nn=513452
+this link leads to all of the certs before 2010, and gather links to all of their pdfs
+"""
+
+test_url = "https://www.bsi.bund.de/SharedDocs/Zertifikate_CC/CC/Digitale_Signatur_Kartenlesegeraete/1046.html" \
+           ";jsessionid=AF4A9C0B7D27992808B8EA8C426454BE.internet461?nn=513452 "
+
+root_url = "https://www.bsi.bund.de/EN/Topics/Certification/certified_products/digital_signature" \
+           "/digital_signature_node.html "
+
+
+"""
+Class used to browse BSI webpage -> get link to all categories
+This class will search for every category, and create a handler for each that will
+retrieve the links to the products
+"""
+
+
+class BsiBrowser(ComplexSerializableType):
+    url: str
+    handler_list: list
+    link_list: list
+
+    def __init__(self, url):
+        self.url = url
+        self.soup = BeautifulSoup(requests.get(self.url).content, "html.parser")
+        self.link_list = []
+        self.handler_list = []
+        self.archive_list = []
+
+    def parse(self):
+
+        self.handler_list = \
+            [BsiHandler("https://www.bsi.bund.de/" + a['href'])
+             for a in self.soup.find_all('a', href=True, recursive=True)
+             if 'c-navigation-teaser' in str(a.get('class'))]
```

Comment:
You can search directly by link class with `find_all(class='c-navigation-teaser')`